### PR TITLE
add different message ids for VSAM errors

### PR DESCRIPTION
**Pull request by taban03**
Addresses https://github.com/zowe/api-layer/issues/3329 

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [ ] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [ ] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes 

#### Changes proposed in this PR

- 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

File: bin/commands/init/vsam/index.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ vsam_volume=
 if [ "${vsam_mode}" = "NONRLS" ]; then
   vsam_volume=$(read_yaml "${ZWE_CLI_PARAMETER_CONFIG}" ".zowe.setup.vsam.volume")
   if [ -z "${vsam_volume}" ]; then
-    print_error_and_exit "Error ZWEL0157E: Zowe Caching Service VSAM data set Non-RLS volume (zowe.setup.vsam.volume) is not defined in Zowe YAML configuration file." "" 157
+    print_error_and_exit "Error ZWEL0148E: Zowe Caching Service VSAM data set Non-RLS attribute (zowe.setup.vsam.volume) is not defined in Zowe YAML configuration file." "" 148
```
Comment (dkelosky) [inline bin/commands/init/vsam/index.sh]:
```suggestion
    print_error_and_exit "Error ZWEL0148E: Zowe Caching Service VSAM data set volume (zowe.setup.vsam.volume) is not defined in Zowe YAML configuration file." "" 148
```

File: bin/commands/init/vsam/.errors [hunk 2]
```diff
@@ -1,3 +1,5 @@
+ZWEL0148E|148|%s (%s) is not defined in Zowe YAML configuration file.
```
Comment (dkelosky) [inline bin/commands/init/vsam/.errors]: it's interesting to see the error message template implemented this way... given that it is the exact same as 149 & 157, it makes me wonder if they should indeed be the exact same message. what do you think?

Comment (taban03) [reply bin/commands/init/vsam/.errors:1]: Yes, I was also concerned about that, and maybe you're right, they could be merged into a single message. Although is also true that the causes are different for all the 3 messages.

Comment (taban03) [reply bin/commands/init/vsam/.errors]: But the failure cause is different, isn't it? I guess that it makes sense to keep them under different msg ID, but maybe I miss something

Comment (MarkAckert) [reply bin/commands/init/vsam/.errors]: I agree with Dan, if we have any "property is missing in zowe.yaml" error, I'd expect the message to share the same ID, with specific error details in the output. If it's a post-processing or derived property error I think there's an argument for that case to have it's own ID.

Comment (Martin-Zeithaml) [reply bin/commands/init/vsam/.errors]: 1) Agree with @dkelosky - this is not a good practice, there is already an error message 157, which is generic and can be used for different problems.
2) This pull request [#3718](https://github.com/zowe/zowe-install-packaging/pull/3718) will remove the shell version  of this command anyway.

Comment (taban03) [reply bin/commands/init/vsam/.errors]: Thanks @Martin-Zeithaml  for the notice about the other PR. Do you know how these messaged will be handled by these changes? Cause I will revert now the changes, but I'm not sure how these kind of messages like:
"Error ZWEL0157E: Zowe Caching Service VSAM data set Non-RLS volume (zowe.setup.vsam.volume) is not defined in Zowe YAML configuration file.
will be then displayed

Comment (Martin-Zeithaml) [reply bin/commands/init/vsam/.errors]: [Like this](https://github.com/zowe/zowe-install-packaging/blob/2188d7db55adde994b5677da01615e81f771001c/bin/commands/init/vsam/index.ts#L45-L51).

**Review by MarkAckert — APPROVED** [review]

**Comment by MarkAckert** [comment]
Approved and merging in knowing #3718 will address this longer term.

**Pull request merged**